examples/apply/apply_bp/apply_bp.jl

```diff
@@ -52,26 +49,26 @@ function simple_update_bp(
 )
   println("Simple update, BP")
   ψψ = norm_network(ψ)
-  mts = message_tensors(partition(ψψ, group(v -> v[1], vertices(ψψ))))
+  pψψ = PartitionedGraph(ψψ, group(v -> v[1], vertices(ψψ)))
   mts = belief_propagation(
-    ψψ, mts; contract_kwargs=(; alg="exact"), niters=50, target_precision=1e-5
+    pψψ; contract_kwargs=(; alg="exact"), niters=50, target_precision=1e-5
   )
+  edges = PartitionEdge.(NamedGraphs.edges(partitioned_graph(pψψ)))
   for layer in eachindex(os)
     @show layer
     o⃗ = os[layer]
     for o in o⃗
       v⃗ = neighbor_vertices(ψ, o)
-      for e in edges(mts)
+      for e in edges
         @assert order(only(mts[e])) == 2
         @assert order(only(mts[reverse(e)])) == 2
       end
 
       @assert length(v⃗) == 2
       v1, v2 = v⃗
 
-      s1 = find_subgraph((v1, 1), mts)
-      s2 = find_subgraph((v2, 1), mts)
-      envs = get_environment(ψψ, mts, [(v1, 1), (v1, 2), (v2, 1), (v2, 2)])
+      pe = partitionedge(pψψ, NamedEdge((v1, 1) => (v2, 1)))
```

Comment:
```suggestion
      pe = partitionedge(pψψ, edgetype(pψψ)((v1, 1) => (v2, 1)))
```
Same comment as below, ideally `partitionedge` could accept either an `AbstractEdge` or a `Pair` as the second input.